lib/puppet/provider/zabbix_hostgroup/ruby.rb

```diff
@@ -2,14 +2,32 @@
 Puppet::Type.type(:zabbix_hostgroup).provide(:ruby, parent: Puppet::Provider::Zabbix) do
   confine feature: :zabbixapi
 
+  def self.instances
+    api_hostgroups = zbx.hostgroups.all
+    api_hostgroups.map do |group_name, _id|
+      new(
+        ensure: :present,
+        name: group_name
```

Comment:
Is `provider: :ruby` needed here?
Or is that just what I’ve done in the past by blindly following what I read somewhere once?! (Probably http://garylarizza.com/blog/2013/12/15/seriously-what-is-this-provider-doing/)